keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/handlers/SlackNotificationHandler.kt

```diff
@@ -11,12 +11,19 @@ import com.netflix.spinnaker.keel.slack.SlackNotificationEvent
 interface SlackNotificationHandler <T: SlackNotificationEvent> {
   val type: NotificationType
 
-  fun sendMessage(notification: T)
-
-  fun  <T : SlackNotificationEvent> Collection<SlackNotificationHandler<*>>.supporting(type: NotificationType) : SlackNotificationHandler<*>? {
-    return find {
-     it.type == type
-    } as? SlackNotificationHandler<*>
-  }
+  fun sendMessage(notification: T, channel: String)
 
 }
+
+fun <T : SlackNotificationEvent> Collection<SlackNotificationHandler<*>>.supporting(
+  type: NotificationType
+): SlackNotificationHandler<T>? =
+  this.find { it.type == type } as? SlackNotificationHandler<T>
+
+
+fun <T : SlackNotificationEvent> Collection<SlackNotificationHandler<*>>.supporting(
+  type: Class<T>
+): SlackNotificationHandler<T>? =
+  this.find { handler ->
+    handler.javaClass == type }
+    as? SlackNotificationHandler<T>
```

Comment:
```suggestion
  this.find { it.javaClass == type } as? SlackNotificationHandler<T>
```

Reply:
I removed it for now as I'm not using it in the end :)